packages/api/lib/writeRecords/write-granules.js

```diff
@@ -483,13 +494,25 @@ const _writeGranuleRecords = async (params) => {
         apiGranuleRecord.createdAt = pgGranule.created_at.getTime();
       }
 
+      if (writeConstraints && (pgGranule.status === 'running' || pgGranule.status === 'queued')) {
+        // pgGranule was updated, but with writeConstraints conditions, so only some values were
+        // updated. we need to ensure the correct values are propogated to Dynamo and ES
+        apiGranuleRecord = await translatePostgresGranuleToApiGranule({
```

Comment:
I don't think we should reassign `apiGranuleRecord`. 

Separately, the `writeConstraints` only apply to the PG writes? It seems like we're saying if we've written with writeConstraints, we've written a different record to PG and Dynamo? I'm guessing yes because the Dynamo write below isn't new. We're just using the PG result instead of the passed in API Granule... I guess. that makes sense. Have to think about that.

Reply:
Under very specific conditions with the two statuses, it will only overwrite status, timestamp, updated_at, and created_at (see `packages/db/src/models/granule.ts:ln128`). So anything else will be ignored, for PG. So I think we need to align what gets written to Dynamo and ES with that. If we are not going to reassign `apiGranuleRecord`, then I think the only option is to make a new translated record, then call the other two writes with that. Seems kind of verbose but I suppose would be _a bit_ more clear. Does that seem acceptable?

Reply:
It seems like we take in the `apiGranuleRecord` param but we don't really use it. We do this conditional reassignment:

```
if (!apiGranuleRecord.createdAt) {
        apiGranuleRecord.createdAt = pgGranule.created_at.getTime();
      }
```

but then we don't use the updated `apiGranuleRecord`, we immediately reassign it?

Reply:
So that conditional was to ensure the PG created_at value got propagated correctly. Now that you point it out I think we can keep that with the original logic, and do the Dynamo and ES writes with a different record if the PG write result requires it.